kobo/apps/hook/models/service_definition_interface.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from abc import ABCMeta, abstractmethod
+import logging
+import json
+import re
+
+import requests
+from django.conf import settings
+from rest_framework import status
+
+from ..constants import HOOK_LOG_SUCCESS, HOOK_LOG_FAILED
+from .hook_log import HookLog
+
+
+class ServiceDefinitionInterface(object):
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, hook, uuid):
+        self._hook = hook
+        self._uuid = uuid
+        self._data = self._get_data()
+
+    def _get_data(self):
+        """
+        Retrieves data from deployment backend of the asset.
+        """
+        try:
+            return self._hook.asset.deployment.get_submission(self._uuid, self._hook.export_type)
+        except Exception as e:
+            logger = logging.getLogger("console_logger")
+            logger.error("service_json.ServiceDefinition._get_data - Hook #{} - Data #{} - {}".format(
+                self._hook.uid, self._uuid, str(e)), exc_info=True)
+
+        return None
+
+    @abstractmethod
+    def _prepare_request_kwargs(self):
+        """
+        Prepares params to pass to `Requests.post` in `send` method.
+        It defines headers and data.
+
+        For example:
+            {
+                "headers": {"Content-Type": "application/json"},
+                "json": self._data
+            }
+        :return: dict
+        """
+        pass
+
+    def send(self):
+        """
+        Sends data to external endpoint
+        :return: bool
+        """
+
+        success = False
+        if self._data:
+            try:
+                request_kwargs = self._prepare_request_kwargs()
+
+                # Add custom headers
+                request_kwargs.get("headers").update(self._hook.settings.get("custom_headers", {}))
+
+                # Add user agent
+                request_kwargs.get("headers").update({
+                    "User-Agent": "KoBoToolbox external service #{}".format(self._hook.uid)
```

Comment:
I'm glad you thought to identify us this way. Now that you've given an inch, let me take a mile: can we include the hostname of the running instance here somehow? It'd be nice to distinguish not only between our instances, but between ours and those run by our many adoring followers.

(not blocking)

Reply:
Trying to get the `hostname` from the container always return container name (i.e. `kpi`)
Because it sends from Celery, we don't have access to `Django.request` object. So ` request.get_host()` is useless. 
We may be able to pass something from the composer files as an `env` variable.

Reply:
According to our discussion. Public domain name was added in User Agent.

https://github.com/kobotoolbox/kpi/commit/232a9f89187c68da8f753ffe121db9edde918814